tests/test_health.py

```diff
@@ -667,20 +667,50 @@ def test_positive_check_postgresql_checkpoint_segments(ansible_module):
         1. foreman-maintain should be installed.
 
     :steps:
-        1. Add config_entries section in /etc/foreman-installer/custom-hiera.yaml
+        1. Have an invalid /etc/foreman-installer/custom-hiera.yaml file
         2. Run foreman-maintain health check --label check-postgresql-checkpoint-segments.
-        3. Assert that check-postgresql-checkpoint-segments fails.
-        4. Add checkpoint_segments parameter in /etc/foreman-installer/custom-hiera.yaml
+        3. Assert that check-postgresql-checkpoint-segments gives proper
+           error message saying an invalid yaml file
+        4. Add config_entries section in /etc/foreman-installer/custom-hiera.yaml
         5. Run foreman-maintain health check --label check-postgresql-checkpoint-segments.
         6. Assert that check-postgresql-checkpoint-segments fails.
-        7. Remove config_entries section from /etc/foreman-installer/custom-hiera.yaml
+        7. Add checkpoint_segments parameter in /etc/foreman-installer/custom-hiera.yaml
         8. Run foreman-maintain health check --label check-postgresql-checkpoint-segments.
-        9. Assert that check-postgresql-checkpoint-segments pass.
+        9. Assert that check-postgresql-checkpoint-segments fails.
+        10. Remove config_entries section from /etc/foreman-installer/custom-hiera.yaml
+        11. Run foreman-maintain health check --label check-postgresql-checkpoint-segments.
+        12. Assert that check-postgresql-checkpoint-segments pass.
+
+    :BZ: 1894149
 
     :expectedresults: check-postgresql-checkpoint-segments should work.
 
     :CaseImportance: High
     """
+    # Create invalid yaml file
+    ansible_module.lineinfile(
+        path="/etc/foreman-installer/custom-hiera.yaml",
+        regexp="---",
+        line="----",
+        state="present",
+    )
+    contacted = ansible_module.command(
+        Health.check(["--label", "check-postgresql-checkpoint-segments"])
+    )
+    for result in contacted.values():
+        logger.info(result["stdout"])
+        assert (
+            "File /etc/foreman-installer/custom-hiera.yaml is not a yaml file." in result["stdout"]
```

Comment:
you can define a variable with the path of `custom-hiera` file and can use it across test

Reply:
Okay.